yt/frontends/cf_radial/data_structures.py

```diff
@@ -0,0 +1,315 @@
+                            storage_filename,
+                        )
+                        mylog.warning(
+                            "To re-grid, either provide a unique storage_filename or set "
+                            "storage_overwrite to True to overwrite %s.",
+                            storage_filename,
+                        )
+                        regrid = False
+                    else:
+                        mylog.info(
+                            "loading existing re-gridded file: %s", storage_filename
+                        )
+                        regrid = False
+
+                if regrid:
+                    mylog.info("Building cfradial grid")
+                    from yt.utilities.on_demand_imports import _pyart as pyart
+
+                    radar = pyart.io.read_cfradial(filename)
+
+                    grid_limit_z = self._validate_grid_dim(radar, "z", grid_limit_z)
+                    grid_limit_x = self._validate_grid_dim(radar, "x", grid_limit_x)
+                    grid_limit_y = self._validate_grid_dim(radar, "y", grid_limit_y)
+                    grid_limits = (grid_limit_z, grid_limit_y, grid_limit_x)
+                    grid_shape = self._validate_grid_shape(grid_shape)
+                    # note: grid_shape must be in (z, y, x) order.
+
+                    self.grid_shape = grid_shape
+                    self.grid_limits = grid_limits
+                    mylog.info("Calling pyart.map.grid_from_radars ... ")
+                    # this is fairly slow
+                    grid = pyart.map.grid_from_radars(
+                        (radar,),
+                        grid_shape=self.grid_shape,
+                        grid_limits=self.grid_limits,
+                    )
+                    mylog.info(
+                        "Successfully built cfradial grid, writing to %s",
+                        storage_filename,
+                    )
+                    mylog.info(
+                        "Subsequent loads of %s will load the gridded file by default",
+                        filename,
+                    )
+                    grid.write(storage_filename)
+
+                filename = storage_filename
+
+        super().__init__(filename, dataset_type, units_override=units_override)
+        self.storage_filename = storage_filename
+        self.refine_by = 2  # refinement factor between a grid and its subgrid
+
+    @contextlib.contextmanager
+    def _handle(self, filename: Optional[str] = None):
+        if filename is None:
+            if hasattr(self, "filename"):
+                filename = self.filename
+            else:
+                raise RuntimeError("Dataset has no filename yet.")
+
+        with xr.open_dataset(filename) as xrds:
+            yield xrds
+
+    def _validate_grid_dim(
+        self, radar, dim: str, grid_limit: Optional[Tuple[float, float]] = None
+    ) -> Tuple[float, float]:
+        if grid_limit is None:
+            if dim == "z":
```

Comment:
will this always be lowercase?  I suspect so, but want to check.

Reply:
I believe, if the files follow the cfradial conventions, they should always be lowercase.

Reply:
sounds good!